md2conf/converter.py

```diff
@@ -115,20 +125,64 @@ def _change_ext(path: str, target_ext: str) -> str:
 class ConfluenceStorageFormatConverter(NodeVisitor):
     "Transforms a plain HTML tree into the Confluence storage format."
 
+    path: str
     base_path: str
     links: List[str]
     images: List[str]
+    page_metadata: Dict[str, ConfluencePageMetadata]
 
-    def __init__(self, base_path: str) -> None:
+    def __init__(self, path: str, page_metadata: Dict[str, ConfluencePageMetadata] = dict()) -> None:
         super().__init__()
-        self.base_path = base_path
+        self.path = path
+        self.base_path = os.path.dirname(path)
         self.links = []
         self.images = []
+        self.page_metadata = page_metadata
 
     def _transform_link(self, anchor: ET.Element) -> ET.Element:
         url = anchor.attrib["href"]
         if not is_absolute_url(url):
+            LOGGER.debug(f"not is_absolute_url = {url} from = {self.path}")
             self.links.append(url)
+            # Convert the relative href to absolute based on relative url the base path value then
+            # look up the absolute path in the page metadata dictionary to discover
+            # the relative path within confluence that should be used
+            abs_path = os.path.abspath(
+                os.path.join(os.path.abspath(self.base_path), url)
+            )
+
+            # Transform relative urls like "#page_anchor" to "$pagepath#page_anchor"
+            if url.startswith("#"):
+                abs_path = self.path + url
+
+            # The path prior to the page anchor if it exists
+            page_path = abs_path.split("#")[0]
```

Comment:
Have you considered the Python module [urllib.parse](https://docs.python.org/3/library/urllib.parse.html)? It may be a more straightforward (and future-proof) solution to split URLs and then re-construct URLs from components than string manipulation, which can have its caveats.